# Functional characterization of the microbiota in MSI-H patients treated with ICI

__Masters Research Internship__  <br>
__Duration:__ Sept 2022 - Feb 2022 
<br>



__Author:__ Miguel Parra Mart√≠nez <br>
__Internship Supervisor:__ Dr. Lodewyk Wessels <br>
__Daily Supervisor:__ Thomas Battaglia <br>
__Masters Program:__ MSc Bioinformatics and Systems Biology (Vrije Universiteit van Amsterdam & University of Amsterdam) <br>
__Host Institute:__ Netherlands Cancer Institute <br>
__Research Group:__ Computational cancer biology <br>
__Focus:__ Microbial Metagenomics  <br>

__Abstract__ <br>
The development of immunotherapy has revolutionized the treatment of cancer, leading to unprecedented clinical responses in some patients. Nevertheless, not all patients respond to these treatments, highlighting the necessity for a better understanding of their mechanism. In recent years several studies have pointed out to the gut microbiome as one of the most important modulators in the response to immunotherapy. Targeting the gut microbiome has attracted much attention as it is simpler than other alternatives, however, the underlying reasons for its effect in immunotherapy are still unknown. Different studies have identified different species of bacteria as the ones responsible for the observed effects with little or no concordance between them. This has emphasized the need to research beyond the taxonomic paradigm and focus more on the functional potential of the gut microbiome as a whole. In this research, we use state-of-the-art functional profiling metagenomic tools to infer the differences in the gut microbiota between responders and non-responders in a pan-cancer cohort (n = 148) of MSI-H patients treated with Nivolumab (anti-PD-1 treatment). Using an assembly-based pipeline we found microbial genes involved in the synthesis of colibactin (vitamin B12) increased in responders and an increment in transmembrane transport processes in non-responders. We also observed that a significant part of the gut microbiome gene products are unknown or uncharacterized, putting into relevance how much is still unknown even in a very deeply studied environment such as the human gut. Lastly, we compare the obtained results with a mapping-based approach, showing that both methods are compatible with each other and provide additive information. In conclusion, this research has provided valuable insight into the potential and limitations of functional gut metagenomics profiling. Nevertheless, the altered gut microbiota processes found in responders and non-responders to immunotherapy should be validated in more detail by metabolomic analysis and wet-lab experiments.

The following workflow was followed in the project research: 

![workflow](https://github.com/miparrama/Internship-NKI/edit/master/workflow.png?raw=true)


