---
title: "Funtional profiling (Metawibele)"
author: "Miguel Parra"
date: "2022-11-07"
output: html_document
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, include=FALSE}
#Import libraries
library (ggplot2)
library("readr")
library("dplyr")
library ("tidyr")
library ("ggrepel")
library ("purrr")
library ("vegan")
library ("ape")
library ("compositions")
library ("robcompositions")
library ("ggfortify")
library ("edgeR")
library ("forcats")
library ("ggsci")
library ("umap")
```


##Loading the data

```{r}
### Characterize outputs: 
annotation <- read_tsv("/home/m.p.martinez/fun_profiling/metawibele/output/run1/characterization2/finalized/metawibele_proteinfamilies_annotation.tsv")
annotation_attribute <- read_tsv ("/home/m.p.martinez/fun_profiling/metawibele/output/run1/characterization2/finalized/metawibele_proteinfamilies_annotation.attribute.tsv") #Supplements the annotation information

### Prioritize outputs: 
prioritize_selected_sup_table <- read_tsv ("/home/m.p.martinez/fun_profiling/metawibele/output/run1/prioritization2/metawibele_supervised_prioritization.rank.selected.table.tsv")
prioritize_selected_sup <- read_tsv ("/home/m.p.martinez/fun_profiling/metawibele/output/run1/prioritization2/metawibele_supervised_prioritization.rank.selected.tsv")
prioritize_selected_unsup_table <- read_tsv ("/home/m.p.martinez/fun_profiling/metawibele/output/run1/prioritization2/metawibele_unsupervised_prioritization.rank.selected.table.tsv")

annotation_attribute <- annotation_attribute %>% separate (col = AID, into = c ("familyID", "category"), sep = "__") # This takes long...

abundance_prot_families <- read_tsv ("/home/m.p.martinez/fun_profiling/metawibele/output/run1/characterization2/finalized/metawibele_proteinfamilies_nrm.tsv")

drup_metadata <- read_csv ("/DATA/share/tom_masters/miguel/drup-gut-metadata.csv")

```
```{r}
path_plots <- "/home/m.p.martinez/fun_profiling/plots/metawibele"
```
Path to sabe the plots. 


##Exploratory data analysis. 

```{r}
#Number of protein clusters: 
n_clusters <- n_distinct (annotation$familyID) #1670180 

#Mean number of proteins that form the clusters
# 1.21 Most of the clusters are composed of only 1 predicted protein. 
annotation_attribute %>% filter (key == "cluster_size") %>% mutate (value = as.numeric(value)) %>% summarise(mean = mean(value), sd = sd(value), min = min(value), max(value), n = n())

annotation_attribute %>% filter (key == "cluster_size") %>% mutate (value = as.numeric (value)) %>% filter (value == 1) %>% nrow() /n_clusters

annotation %>% distinct(method) %>% pull()

```
Most of the clusters (87%) are formed by 1 protein only. In line with Metawibele publication (2023587 predicted proteins in origin)


### Taxonomy annotation
```{r}
annotation_taxonomy <- annotation %>% filter (method == "Taxonomy_annotation")

annotation_taxonomy  %>% group_by (feature) %>% summarise (n = n(), freq = n()/n_clusters)

annotation_taxonomy  %>% 
  ggplot (aes (feature)) + geom_bar(aes (fill = feature), show.legend = FALSE) + scale_y_log10() + labs (y = "log(No.) of proteins families", x = "Taxonomic classification")

annotation_taxonomy %>% filter (feature == "Unclassified") %>% nrow() / n_clusters


## Most of the predicted proteins are unclassified 
```
Most of the predicted proteins are unclassified. (96%)

This contrast with the paper of timetable which shows that most of the proteins can be taxonomically classified. 

```{r}
annotation %>% filter (method == "MSPminer") %>% select (annotation) %>% distinct()

```

This is the number of pangenomes generated. 



### Homology
```{r}
annotation_uniref <- annotation %>% filter (method == "UniRef90") %>% select (-c (method,AID))
annotation_uniref %>% head()

annotation_uniref %>% filter (category == "UniRef90_characterization") %>% ggplot (aes (feature)) + geom_bar() + coord_flip()



annotation_uniref %>% filter (category == "UniRef90_homology") %>% group_by(feature) %>% summarise(count = n())
annotation_uniref %>% filter (category == "UniRef90_homology") %>%
  ggplot (aes (feature)) + geom_bar(aes (fill = feature), show.legend = FALSE) +
  labs (x = " Homology against UniRef90", y = "No. of protein families") +
  scale_fill_manual (values = c ("#52854C", "#4E84C4", "#D16103"))

annotation_uniref %>% filter (category == "UniRef90_homology") %>% group_by (feature) %>% summarise( count = n(), percentage = n()/n_clusters)
#Most of the proteins have a strong uniref90 homology. 

```
Most of the proteins have a strong uniref90 homology.



```{r}
abundance_prot_families <- abundance_prot_families %>% rename (familyID = ID)
homology_abun_per_sample <- annotation_uniref %>% filter (category == "UniRef90_homology") %>% select (familyID, feature) %>% left_join(abundance_prot_families)


homology_abun_per_sample3 <- annotation_uniref %>% filter (category == "UniRef90_characterization") %>% select (familyID, feature) %>% rename (feature2 = feature) %>% filter (feature2 %in% c ("UniRef90_unknown", "UniRef90_uncharacterized")) %>% right_join(homology_abun_per_sample) %>% mutate ( feature2 = replace_na(feature2, "UniRef90_characterized")) %>%  mutate (homology = case_when(
  feature2 == "UniRef90_uncharacterized" & feature == "strong_homology" ~ "SU", 
  feature2 == "UniRef90_characterized" & feature == "strong_homology" ~ "SC", 
  feature2 == "UniRef90_unknown" & feature == "weak_homology" ~ "RH", 
  feature2 == "UniRef90_unknown" & feature == "worse_homology" ~ "NH"
)) %>% select (- c(feature2, feature)) %>% relocate (homology, .after = 1)


homology_abun_per_sample2 <- homology_abun_per_sample %>% group_by(feature) %>% summarize_at(vars(-1), sum, na.rm = TRUE)  %>% pivot_longer(, cols = -feature, names_to = "sample", values_to = "abundance")

homology_abun_per_sample2 <- homology_abun_per_sample3 %>% group_by(homology) %>% summarize_at( vars (-c("familyID", "mean_abun", "prevalence")), sum, na.rm = TRUE)  %>% pivot_longer(, cols = -homology, names_to = "Sample", values_to = "abundance") %>% left_join((drup_metadata %>% select (c (Sample, Response))))



# Stacked + percent
ggplot(homology_abun_per_sample2, aes(fill=homology, y=abundance, x=Sample)) + 
    geom_bar(position="fill", stat="identity", width = 1) + 
  theme(axis.text.x  = element_blank()) + 
  labs (y = "Abundance(%)", fill = "Homology") +
  theme(text = element_text(size=14))
ggsave ("abun_percent_homology_sample.png", path = path_plots)



homology_abun_per_sample2 %>% filter (Response %in% c("Responder", "Non-responder")) %>% ggplot( aes(fill=homology, y=abundance, x=Sample)) + 
    geom_bar(position="fill", stat="identity", width = 1) + 
  theme(axis.text.x  = element_blank()) + 
  labs (y = "Abundance(%)", fill = "Homology") + 
  facet_grid( cols = vars(Response), scales="free_x") +
  theme(text = element_text(size=16))
ggsave ("abun_percent_homology_sample_facet.response.png", path = path_plots)



## Distribution of mean abundance of protein families. 

cluster_mean_abun <- homology_abun_per_sample3 %>% select (- c(familyID, homology)) %>% rowMeans()
homology_abun_per_sample3 <- homology_abun_per_sample3 %>% mutate (mean_abun = cluster_mean_abun)
ggplot (homology_abun_per_sample3, aes (x = log(mean_abun), fill = homology, y = stat(count))) + geom_density (alpha=.3) + 
  theme(text = element_text(size=16))
ggsave ("mean_abundance_homology.png", path = path_plots)

```


```{r}

## Number of proteins forminf the cluster depending on the homology...
cluster_size <- annotation_attribute %>% filter (key == "cluster_size") 

homology_abun_per_sample3["cluster_size"] <- cluster_size %>% mutate(value = as.numeric(value)) %>% pull(value)

homology_abun_per_sample3 %>% group_by(homology) %>% summarise (mean = mean (cluster_size), max = max(cluster_size))

```



*Mean prevalence*

```{r}
# Add prevalence to homology_abun_per_sample3
homology_abun_per_sample3 <- homology_abun_per_sample3 %>% left_join(., (annotation %>% filter( feature == "DNA_prevalence") %>% select (familyID, annotation) %>% rename (prevalence = annotation))) 

#prevalence plot (against abundance...)
homology_abun_per_sample3 %>% drop_na() %>% ggplot (aes (x = prevalence, y = log(mean_abun))) + geom_jitter(alpha = 0.05) + theme(text = element_text(size=14))

homology_abun_per_sample3 %>%  drop_na() %>% mutate (prevalence = jitter (as.double(prevalence), factor = 2)) %>%  ggplot (aes (x = prevalence, y = log(mean_abun)), position = "jitter") + geom_hex(binwidth = c(0.01, 0.16), aes(fill = stat(count))) + facet_grid(rows = vars(homology)) + theme(text = element_text(size=14))
ggsave(file="log.abun_prevalence_facet.homology.png", path = path_plots, width=6, height=8, dpi=300)

#Density of prevalence
homology_abun_per_sample3 %>%  drop_na() %>% mutate (prevalence = jitter (as.double(prevalence), factor = 2)) %>% ggplot (aes (x = prevalence, fill = homology,  y = stat(count))) + geom_density (alpha=.3) + theme(text = element_text(size=14))
ggsave(file="density_prevalence.png", path = path_plots)


#Make prevalence one depending on responden or non responders....

```



```{r}
#Number of proteins families per each 
ggplot (homology_abun_per_sample3, aes (homology)) + geom_bar( aes(fill = homology)) + 
  theme(text = element_text(size=14))

### Plot the proportions as well. 
ggsave ("count_homology.png", path = path_plots)


#Proportions or each cluster by homology
homology_abun_per_sample3 %>% group_by(homology) %>% summarise (per = round (100*n()/n_clusters, 2), n = n())



```

distribution of protein lengths between the groups of protein families. 
```{r}
cluster_length <- annotation_attribute %>% filter (key == "rep_length") %>% select ( c("familyID", "value")) %>% rename (length = value) %>% mutate (length = as.numeric(length))

homology_abun_per_sample3 <- homology_abun_per_sample3 %>% left_join(cluster_length)

homology_abun_per_sample3 %>% ggplot (aes (x = log(length), fill = homology, y = stat(count))) + geom_density(alpha=.3) + 
  theme (text = element_text(size = 14)) 
ggsave ("protein_lenght_distribution.png", path = path_plots)






```
```{r}


n_clusters_per_sample_per_homology <- homology_abun_per_sample3 %>% select (-c("mean_abun", "prevalence", "cluster_size", "length")) %>% pivot_longer(cols = contains ("DRUP"), names_to = "sampleId", values_to = "abun") %>% filter (abun > 0 ) %>% group_by(sampleId, homology) %>% summarise(n = n ())

n_clusters_per_sample_per_homology %>% 
  ggplot (aes (x = sampleId, y = n, fill = homology)) + 
  geom_bar ( stat = "identity", width = 1) +
  theme(axis.text.x  = element_blank(), text = element_text(size = 14)) + 
  labs (y = "# of protein clusters", fill = "Homology")


ggsave ("numb_clusters_homology_sample.png", path = path_plots)


```




```{r}
annotation_uniref %>% filter (category == "UniRef90_characterization") %>%
  #filter (!c("UniRef90_uncharacterize", "UniRef90_unknown")) %>% 
  group_by(feature) %>% summarise (n = n(), per = round (100*n()/340913, 2))
  #filter (feature == "UniRef90_PfamDomain") %>% 
  #distinct(familyID) %>% nrow()
  
```
More protein families with UniRef90_PfamDomain than StrongHomology...




###Functial annotation exploration

```{r}
annotation %>% distinct(method) %>% pull()
```





*TMHMM/Phobius*. Transmembrane regions. 
```{r}
annotation %>% filter (method == "TMHMM/Phobius") %>% group_by(feature) %>% summarise (n =n(), percentage = round (100 * n() / n_clusters, digits = 3))
```



*Phobius* and *SignalP* are tools that identify signaling peptides in protein sequences. 
SignalP uses deep neural networks, while Phobius is based in HMM. 
```{r}
#SignalP and Phobius identigy the signaling peptides. 
annotation %>% filter (method == "SignalP/Phobius") %>% group_by(feature) %>% summarise (n =n(), percentage = round (100 * n() / n_clusters, digits = 3))
##8.37% of the proteins clusters are predicted to have a signaling domain. 

```




*Domine* is a database of known and predicted protein domain (domain-domain) interactions. It contains interactions inferred from PDB entries, and those that are predicted by 13 different computational approaches using Pfam domain definitions.
There is also information about the specific hits (DDI) with human proteins
```{r}

#DOMINE
annotation %>% filter (method == "DOMINE") %>% group_by(feature) %>% summarise (n =n(), percentage = round (100 * n() / n_clusters, digits = 3))
```
Few proteins have this annotation... (0.4%). 


*ExpAtlas* or Expression Atlas database provides information about gene expression levels from RNA-seq studies and Microarrays studies. 
```{r}
#ExpAtlas
annotation %>% filter (method == "ExpAtlas") %>% group_by(feature) %>% summarise (n =n(), percentage = round (100 * n() / n_clusters, digits = 3))
```
The same protein families that have a Domine annotation also have a expression atlas annotation. 

*SIFTS* 
```{r}
#SIFTS
annotation %>% filter (method == "SIFTS") %>% group_by (feature) %>% summarise (n = n(), percentage = round (100 * n() / n_clusters, 3))
```


*InterProScan* InterProScan combines multiple analysis: 
```{r}
annotation %>% filter (method == "InterProScan") %>% group_by(feature) %>% summarise (n =n(), percentage = round (100 * n() / n_clusters, digits = 3))
```


*PfamDomain* Database of protein families that include annotations. 

```{r}
annotation %>% filter (feature == "InterProScan_PfamDomain") %>% group_by(annotation) %>% summarise (n = n())
```
PF00004 --> ATPasa familiy 
PF09847 --> Membrane protein of 12 TMs    



*Panther* (Protein Analysis Through Evolutionary relationships).
```{r}
annotation %>% filter (feature == "InterProScan_PANTHER") %>% group_by(annotation) %>% summarise (n = n())
```
PTHR23076 --> METALLOPROTEASE M41 FTSH (PTHR23076)


*CDD* (Conserved Domain Database) is a protein annotation resource that consists of a collection of well-annotated multiple sequence alignment models for ancient domains and full-length protein
```{r}
annotation %>% filter (feature == "InterProScan_CDD") %>% group_by(annotation) %>% summarise (n =n())
```
cd00009 --> ATP binding site
cd00958 --> Active site involved in Schiff base formation; forms covalent linkage with the product
cd04242 --> Nucleotide binding site 

*Coils* Prediction of coiled coil regions in proteins.
```{r}
annotation %>% filter (feature == "InterProScan_Coils") %>% group_by(annotation) %>% summarise (n =n())

```


*MobiDBLite*. Tool for the prediciton of intrinsically disordered proteins. 
```{r}
annotation %>% filter (feature == "InterProScan_MobiDBLite") 
```


*Gene 3D* Structural assignment for whole genes and genomes using the CATH domain structure database. 
```{r}
annotation %>% filter (feature == "InterProScan_Gene3D") %>% group_by(annotation) %>% summarise (n = n())

```




## Differential abundance analysis
```{r}

# Generate abundance and log Fold change table. 
abundance_df <- annotation_attribute %>% filter (category %in% c("MaAsLin2_DA", "UniRef90_homology"), key %in% c("qvalue", "log(FC)", "Protein_names")) %>% select (-TID) %>% select(-category)  %>% pivot_wider (names_from = key, values_from = value)
abundance_df <- abundance_df %>% rename( log_FC = `log(FC)`)  %>% mutate (qvalue = as.numeric(qvalue), log_FC = as.numeric(log_FC))

# Nas in qvalue and log(FC) are protein families are not present in one of the conditions. 
abundance_df <- abundance_df %>% drop_na (!Protein_names)

#Filter by minimun of 30% prevalence in both groups. 

clusters_min_prevalence <- annotation %>% filter (feature %in% c( "DNA-Non-responder_prevalence", "DNA-Responder_prevalence")) %>% 
  group_by (familyID) %>%
  distinct (feature, .keep_all = TRUE) %>% ungroup() %>% select (-c(category,method,AID)) %>% #Delete the repeated empty line.
  pivot_wider (names_from = feature, values_from = annotation) %>%
  filter(`DNA-Non-responder_prevalence` > 0.5 & `DNA-Responder_prevalence` > 0.5) #Threshold of prevalence


#Volcano plot XD
p1 <- abundance_df %>% filter (familyID %in% (clusters_min_prevalence %>% pull(familyID))) %>% drop_na (!Protein_names) %>% 
  ggplot(aes ( log_FC, -log(qvalue,10))) + # -log10 conversion  
  geom_point(size = 2/5) +
  xlab(expression("log"[2]*"FC")) + 
  ylab(expression("-log"[10]*"FDR")) +
  theme(text = element_text(size=14))
  
p1


```
The volcano plot is not very convincing (maybe be because is a linear mixed model?) 




```{r}

abundance_prot_families %>% select (-ID) %>% colSums() ## To many numbers to be added ?
# The total number of occurance of proteins is very similar between samples. Min=96e+05, Max=99e+05



# Number or proteins families per sample: 
nprot_families_per_sample <- abundance_prot_families %>% summarise_all(funs(sum(. > 0)))

# Selects only the samples that where filtered. (only responders and non-responders). 
nprot_families_per_sample2 <- nprot_families_per_sample %>% select (samples_metadata_filtered$Sample) %>% t() %>% as_tibble(column_name = "Number_prot_clusters")
nprot_families_per_sample2 <- nprot_families_per_sample2 %>% add_column ( Sample =(nprot_families_per_sample %>% select (samples_metadata_filtered$Sample) %>% colnames()), .before = 1) 

nprot_families_per_sample2 <- rename (nprot_families_per_sample2, N_prots = V1)
 
nprot_families_per_sample2 <- nprot_families_per_sample2 %>% left_join (., (samples_metadata_filtered %>% select (c("Sample", "Response"))))

p <- ggplot(nprot_families_per_sample2, aes (x = Response, y = N_prots)) +
  geom_violin() +
  geom_jitter() +
  ylab("Number protein clusters") + 
  xlab ("") + 
  theme(text = element_text (size = 16))
  
p
ggsave("proteinclusternumber_response.png", path = path_plots)






```

*Rarefracion analysis:*
```{r}
abundance_prot_families_mt <- abundance_prot_families %>% column_to_rownames(var = "ID") %>%  select (samples_metadata_filtered$Sample) %>% as.matrix() 

rarecurve <- rarecurve(round(t(abundance_prot_families_mt)), step = 10000, label = F, tidy = TRUE)

rarecurve <- rarecurve %>% rename (Sample_Size = Sample, Sample = Site) %>% left_join (., (samples_metadata_filtered %>% select (c("Sample", "Response", "patientId")))) 



## Rarecurve plot
rarecurve_plot <- rarecurve %>% ggplot(aes (x = Sample_Size, y = Species, color = Response)) + 
  geom_line(aes (group = Sample), alpha = 0.3) +
  # geom_smooth (aes (group = Response) , size = 2 , method = "loess")
  geom_smooth (data=rarecurve %>% filter (Sample_Size < 960002) %>% group_by(Response,Sample_Size) %>% summarise (mean = mean (Species), sd = sd(Species)),
               aes (x = Sample_Size, y = mean, color = Response, ymin = mean - sd, ymax = mean + sd), stat = "identity", se = FALSE)
rarecurve_plot

```


*Principal Coordinate Analysis.*: 

First applying directly the Bray curtis dissimilarity and plottiing the first two PCoA. 
```{r}

bray_distance <- vegdist (t(abundance_prot_families_mt))
pcoa <- pcoa(bray_distance)

pcoa_plot <- pcoa$vectors %>% as_tibble() %>% add_column(Sample = rownames(pcoa$vectors), .before = 1) %>% left_join (., (samples_metadata_filtered %>% select (c("Sample", "Response", "patientId")))) %>% 
  ggplot(aes ( x = Axis.1, y = Axis.2)) + geom_point (aes (color = Response)) + 
  xlab (paste0("PCo1 (",round(pcoa$values[1,1], 2),"%)"))+
  ylab (paste0("PCo2 (",round(pcoa$values[2,1], 2),"%)")) +
  ggtitle ("PCoA protein clusters abundance")
pcoa_plot
```



Applying Log Centered Ration normalization to the data. 
```{r}
abundance_prot_families_mt_clr <- clr (t(abundance_prot_families_mt))
#euclidean <- vegdist (t(abundance_prot_families_mt_clr))

pca_abundance <- prcomp (abundance_prot_families_mt_clr)


#pca_plot <- autoplot (pca_abundance, data = abun_prot_fam_clr_df, color = "Response")



PCs_abundance <- pca_abundance$x %>% as_tibble() %>%  add_column(Sample = rownames(pca_abundance$x), .before = 1) %>% left_join (., (samples_metadata_filtered %>% select (c("Sample", "Response", "patientId", "timepoint", "Batch", "tumor_type", "BOR", "CB", "ATB_use"))))  %>% mutate (, Id_num = as.numeric(factor (PCs_abundance$patientId)))

PCs_12_mean_sample <- PCs_abundance %>% group_by(patientId) %>% summarise (x = mean(PC1), y = mean (PC2)) %>% mutate (, Id_num = as.numeric(factor (.$patientId)))
PCs_abundance <- PCs_abundance %>% left_join (., PCs_12_mean_sample)


#Response
pca_plot1 <- PCs_abundance %>% ggplot (aes (x = PC1, y = PC2)) + geom_point( aes (color = Response)) +
  xlab (paste0("PC1 (",round(100 * pca_abundance$sdev[1] ** 2 / sum (pca_abundance$sdev **2), 2),"%)"))+
  ylab (paste0("PC2 (",round(100 * pca_abundance$sdev[2] ** 2 / sum (pca_abundance$sdev **2), 2),"%)")) +
  #ggtitle ("PCA protein clusters abundance") +
  theme(text = element_text(size=12))

#Timepoint
pca_plot2 <- PCs_abundance %>% ggplot (aes (x = PC1, y = PC2)) + geom_point( aes (color = timepoint)) +
  xlab (paste0("PC1 (",round(100 * pca_abundance$sdev[1] ** 2 / sum (pca_abundance$sdev **2), 2),"%)"))+
  ylab (paste0("PC2 (",round(100 * pca_abundance$sdev[2] ** 2 / sum (pca_abundance$sdev **2), 2),"%)")) +
  #ggtitle ("PCA protein clusters abundance") +
  theme(text = element_text(size=12))

#Batch
pca_plot3 <- PCs_abundance %>% ggplot (aes (x = PC1, y = PC2)) + geom_point( aes (color = Batch)) +
  xlab (paste0("PC1 (",round(100 * pca_abundance$sdev[1] ** 2 / sum (pca_abundance$sdev **2), 2),"%)"))+
  ylab (paste0("PC2 (",round(100 * pca_abundance$sdev[2] ** 2 / sum (pca_abundance$sdev **2), 2),"%)")) +
  #ggtitle ("PCA protein clusters abundance")+
  theme(text = element_text(size=12))

#Tumor_type
pca_plot4 <- PCs_abundance %>% ggplot (aes (x = PC1, y = PC2)) + geom_point( aes (color = tumor_type)) +
  xlab (paste0("PC1 (",round(100 * pca_abundance$sdev[1] ** 2 / sum (pca_abundance$sdev **2), 2),"%)"))+
  ylab (paste0("PC2 (",round(100 * pca_abundance$sdev[2] ** 2 / sum (pca_abundance$sdev **2), 2),"%)")) +
  theme(text = element_text(size=12))

#Patient ID
pca_plot5 <- PCs_abundance %>% ggplot (aes (x = PC1, y = PC2)) + 
  geom_point(aes (color = patientId), show.legend = FALSE) +
  geom_text_repel(data = PCs_12_mean_sample, aes (x = x, y = y, label = Id_num), size = 3) +
  #geom_point (aes (x = x, y = y, color = patientId), show.legend = FALSE) +
  geom_segment(aes(x = x, y = y, xend = PC1, yend = PC2, color = patientId), show.legend = FALSE, alpha = 0.4) +
  xlab (paste0("PC1 (",round(100 * pca_abundance$sdev[1] ** 2 / sum (pca_abundance$sdev **2), 2),"%)"))+
  ylab (paste0("PC2 (",round(100 * pca_abundance$sdev[2] ** 2 / sum (pca_abundance$sdev **2), 2),"%)")) +
  theme(text = element_text(size=12))

#Antibiotics use
pca_plot6 <- PCs_abundance %>% ggplot (aes (x = PC1, y = PC2)) + geom_point( aes (color = ATB_use)) +
  xlab (paste0("PC1 (",round(100 * pca_abundance$sdev[1] ** 2 / sum (pca_abundance$sdev **2), 2),"%)"))+
  ylab (paste0("PC2 (",round(100 * pca_abundance$sdev[2] ** 2 / sum (pca_abundance$sdev **2), 2),"%)")) +
  theme(text = element_text(size=12))

#Clinical Benefit
pca_plot7 <- PCs_abundance %>% ggplot (aes (x = PC1, y = PC2)) + geom_point( aes (color = CB)) +
  xlab (paste0("PC1 (",round(100 * pca_abundance$sdev[1] ** 2 / sum (pca_abundance$sdev **2), 2),"%)"))+
  ylab (paste0("PC2 (",round(100 * pca_abundance$sdev[2] ** 2 / sum (pca_abundance$sdev **2), 2),"%)")) +
  theme(text = element_text(size=12))


pca_plot1
ggsave("pca_abundance_response.png", path = path_plots, height = 4, width = 6)
pca_plot2
ggsave("pca_abundance_timepoint.png", path = path_plots, height = 4, width = 6)
pca_plot3
ggsave("pca_abundance_batch.png", path = path_plots, height = 4, width = 6)
pca_plot4
ggsave("pca_abundance_tumortype.png", path = path_plots, height = 4, width = 6)
pca_plot5
ggsave("pca_abundance_patientid.png", path = path_plots, height = 4, width = 6)
pca_plot6
ggsave("pca_abundance_antibiotics.png", path = path_plots, height = 4, width = 6)
pca_plot7
ggsave("pca_abundance_cb.png", path = path_plots, height = 4, width = 6)


```

```{r}
pca_abundance <- prcomp (abundance_prot_families_mt_clr)
#pca_plot <- autoplot (pca_abundance, data = abun_prot_fam_clr_df, color = "Response")
```


*UMAP*
```{r}

#Calculation of the UMAP
umap_abundance <- umap (as.matrix(abundance_prot_families_mt_clr))

#Generate table with UMAP1 and 2 with other variables for plotting. 
umap_df <- umap_abundance$layout %>% as_tibble() %>%  rename(UMAP1="V1",UMAP2="V2") %>% add_column (Sample = rownames (umap_abundance$layout), .before = 1) %>% left_join(., (samples_metadata_filtered %>% select (c("Sample", "Response", "patientId", "timepoint", "Batch", "tumor_type", "BOR", "CB", "ATB_use") ))) %>% mutate (, Id_num = as.numeric(factor(PCs_abundance$patientId)))


umap_mean_sample <- umap_df %>% group_by(patientId) %>% summarise (x = mean(UMAP1), y = mean (UMAP2)) %>% mutate (, Id_num = as.numeric(factor (.$patientId)))
umap_df <- umap_df %>% left_join (., umap_mean_sample)


#Response
umap_plot1 <- umap_df %>% ggplot (aes (x = UMAP1, y = UMAP2)) + geom_point( aes (color = Response)) +
  #xlab (paste0("UMAP1 (",round(100 * pca_abundance$sdev[1] ** 2 / sum (pca_abundance$sdev **2), 2),"%)"))+
  #ylab (paste0("UMAP2 (",round(100 * pca_abundance$sdev[2] ** 2 / sum (pca_abundance$sdev **2), 2),"%)")) +
  #ggtitle ("UMAP protein clusters abundance")
  theme(text = element_text(size=12))


#Timepoint
umap_plot2 <- umap_df %>% ggplot (aes (x = UMAP1, y = UMAP2)) + geom_point( aes (color = timepoint)) +
  #ggtitle ("UMAP protein clusters abundance")
  theme(text = element_text(size=12))

#Batch
umap_plot3 <- umap_df %>% ggplot (aes (x = UMAP1, y = UMAP2)) + geom_point( aes (color = Batch)) +
  theme(text = element_text(size=12))

#Tumor_type
umap_plot4 <- umap_df %>% ggplot (aes (x = UMAP1, y = UMAP2)) + geom_point( aes (color = tumor_type)) +
  theme(text = element_text(size=12))


#Patient ID
umap_plot5 <- umap_df %>% ggplot (aes (x = UMAP1, y = UMAP2)) + 
  geom_point(aes (color = patientId), show.legend = FALSE) +
  geom_text_repel(data = umap_mean_sample, aes (x = x, y = y, label = Id_num), size = 3) +
  #geom_point (aes (x = x, y = y, color = patientId), show.legend = FALSE) +
  geom_segment(aes(x = x, y = y, xend = UMAP1, yend = UMAP2, color = patientId), show.legend = FALSE, alpha = 0.4) 
  #theme(text = element_text(size=12))


#Antibiotics use
umap_plot6 <- umap_df %>% ggplot (aes (x = UMAP1, y = UMAP2)) + geom_point( aes (color = ATB_use)) +
  theme(text = element_text(size=12))

#Clinical Benefit
umap_plot7 <- umap_df %>% ggplot (aes (x = UMAP1, y = UMAP2)) + geom_point( aes (color = CB)) +
  theme(text = element_text(size=12))


umap_plot1
ggsave("umap_abundance_response.png", path = path_plots, height = 4, width = 6)
umap_plot2
ggsave("umap_abundance_timepoint.png", path = path_plots, height = 4, width = 6)
umap_plot3
ggsave("umap_abundance_batch.png", path = path_plots, height = 4, width = 6)
umap_plot4
ggsave("umap_abundance_tumortype.png", path = path_plots, height = 4, width = 6)
umap_plot5
ggsave("umap_abundance_patientid.png", path = path_plots, height = 4, width = 6)
umap_plot6
ggsave("umap_abundance_antibiotics.png", path = path_plots, height = 4, width = 6)
umap_plot7
ggsave("umap_abundance_cb.png", path = path_plots, height = 4, width = 6)

```


*UMAP* using protein abundance based on the homology classification. 

- No homology: 
```{r}

#Filter abundance for only protein clusters with no homology. 
abundance_prot_nh <- homology_abun_per_sample3 %>% filter (homology == "NH") %>% column_to_rownames("familyID") %>% select (contains("DRUP0")) %>% as.matrix() %>% t() %>% clr()


umap_abundance_nh <- umap (as.matrix(abundance_prot_nh))


umap_df_nh <- umap_abundance_nh$layout %>% as_tibble() %>% rename ("UMAP1" = "V1", "UMAP2" = "V2") %>% add_column (Sample = rownames (umap_abundance_nh$layout), .before = 1) %>% inner_join (., (samples_metadata_filtered %>% select (c ("Sample", "Response", "patientId", "timepoint", "Batch", "tumor_type", "BOR", "CB", "ATB_use"))))
umap_mean_sample_nh <- umap_df_nh %>% group_by(patientId) %>% summarise (x = mean(UMAP1), y = mean (UMAP2)) %>% mutate (, Id_num = as.numeric(factor (.$patientId)))
umap_df_nh <- umap_df_nh %>% left_join (., umap_mean_sample_nh)

#Response
umap_plot1 <- umap_df_nh %>% ggplot (aes (x = UMAP1, y = UMAP2)) + geom_point( aes (color = Response)) +
  theme(text = element_text(size=12))


#Patient ID
umap_plot5 <- umap_df_nh %>% ggplot (aes (x = UMAP1, y = UMAP2)) + 
  geom_point(aes (color = patientId), show.legend = FALSE) +
  geom_text_repel(data = umap_mean_sample_nh, aes (x = x, y = y, label = Id_num), size = 3) +
  #geom_point (aes (x = x, y = y, color = patientId), show.legend = FALSE) +
  geom_segment(aes(x = x, y = y, xend = UMAP1, yend = UMAP2, color = patientId), show.legend = FALSE, alpha = 0.4) 
  #theme(text = element_text(size=12))


#Antibiotics use
umap_plot6 <- umap_df_nh %>% ggplot (aes (x = UMAP1, y = UMAP2)) + geom_point( aes (color = ATB_use)) +
  theme(text = element_text(size=12))



umap_plot1
umap_plot5
umap_plot6


```

Remote homology. 
```{r}
#Filter for remote homology
abundance_prot_homology_filtered <- homology_abun_per_sample3 %>% filter (homology == "RH") %>% column_to_rownames("familyID") %>% select (contains("DRUP0")) %>% as.matrix() %>% t() %>% clr()

umap_abundance <- umap (as.matrix(abundance_prot_homology_filtered))

umap_df <- umap_abundance$layout %>% as_tibble() %>%  rename("UMAP1"="V1", "UMAP2"="V2") %>% add_column (Sample = rownames (umap_abundance$layout), .before = 1) %>% inner_join(., (samples_metadata_filtered %>% select (c("Sample", "Response", "patientId", "timepoint", "Batch", "tumor_type", "BOR", "CB", "ATB_use") ))) 

umap_mean_sample <- umap_df %>% group_by(patientId) %>% summarise (x = mean(UMAP1), y = mean (UMAP2)) %>% mutate (, Id_num = as.numeric(factor (.$patientId)))
umap_df <- umap_df %>% left_join (., umap_mean_sample)


#Response
umap_plot1 <- umap_df %>% ggplot (aes (x = UMAP1, y = UMAP2)) + geom_point( aes (color = Response)) +
  theme(text = element_text(size=12))

#Patient ID
umap_plot5 <- umap_df %>% ggplot (aes (x = UMAP1, y = UMAP2)) + 
  geom_point(aes (color = patientId), show.legend = FALSE) +
  geom_text_repel(data = umap_mean_sample, aes (x = x, y = y, label = Id_num), size = 3) +
  #geom_point (aes (x = x, y = y, color = patientId), show.legend = FALSE) +
  geom_segment(aes(x = x, y = y, xend = UMAP1, yend = UMAP2, color = patientId), show.legend = FALSE, alpha = 0.4) 
  #theme(text = element_text(size=12))


#Antibiotics use
umap_plot6 <- umap_df %>% ggplot (aes (x = UMAP1, y = UMAP2)) + geom_point( aes (color = ATB_use)) +
  theme(text = element_text(size=12))



umap_plot1
umap_plot5
umap_plot6



```



Strong homology. 
```{r}
#Filter for remote homology
abundance_prot_homology_filtered <- homology_abun_per_sample3 %>% filter (homology %in% c ("SC", "SU")) %>% column_to_rownames("familyID") %>% select (contains("DRUP0")) %>% as.matrix() %>% t() %>% clr()

umap_abundance <- umap (as.matrix(abundance_prot_homology_filtered))

umap_df <- umap_abundance$layout %>% as_tibble() %>%  rename("UMAP1"="V1", "UMAP2"="V2") %>% add_column (Sample = rownames (umap_abundance$layout), .before = 1) %>% inner_join(., (samples_metadata_filtered %>% select (c("Sample", "Response", "patientId", "timepoint", "Batch", "tumor_type", "BOR", "CB", "ATB_use") ))) 

umap_mean_sample <- umap_df %>% group_by(patientId) %>% summarise (x = mean(UMAP1), y = mean (UMAP2)) %>% mutate (, Id_num = as.numeric(factor (.$patientId)))
umap_df <- umap_df %>% left_join (., umap_mean_sample)


#Response
umap_plot1 <- umap_df %>% ggplot (aes (x = UMAP1, y = UMAP2)) + geom_point( aes (color = Response)) +
  theme(text = element_text(size=12))

#Patient ID
umap_plot5 <- umap_df %>% ggplot (aes (x = UMAP1, y = UMAP2)) + 
  geom_point(aes (color = patientId), show.legend = FALSE) +
  geom_text_repel(data = umap_mean_sample, aes (x = x, y = y, label = Id_num), size = 3) +
  #geom_point (aes (x = x, y = y, color = patientId), show.legend = FALSE) +
  geom_segment(aes(x = x, y = y, xend = UMAP1, yend = UMAP2, color = patientId), show.legend = FALSE, alpha = 0.4) 
  #theme(text = element_text(size=12))


#Antibiotics use
umap_plot6 <- umap_df %>% ggplot (aes (x = UMAP1, y = UMAP2)) + geom_point( aes (color = ATB_use)) +
  theme(text = element_text(size=12))



umap_plot1
umap_plot5
umap_plot6



```


###Satatistical analysis: Limma
```{r}

colnames (abundance_prot_families_mt)
PCs_abundance %>% pull (Sample) %>% factor ()

plotMDS (abundance_prot_families_mt, col = as.numeric(factor(PCs_abundance$Response)))
plotMDS( abundance_prot_families_mt, col = as.numeric (factor (PCs_abundance$Response)), top = 1000, gene.selection = "common")

```
?? meaning of this compared with the PCA??.. Is PCoA using the distance between samples only of the "top" n genes. The top genes in this case with largest standard variation. 







Needed graphs: 

- Distribution of length of the protein clusters depending on their homology. 
- Number of protein clusters per sample. 
- Why there are more pfam annotations?





